### ISSUE TITLE
Unlisted Microsoft.Management.Infrastructure by default

### ISSUE THREAD

**Issue by theimowski** [fsprojects/Paket#3757]
### Description

Paket resolves Microsoft.Management.Infrastructure to an unlisted version: [4.0](https://www.nuget.org/packages/Microsoft.Management.Infrastructure/4.0.0)

This package *might* be special in regard that this unlisted version is actually higher than the latest supported, which is currently [2.0](https://www.nuget.org/packages/Microsoft.Management.Infrastructure/2.0.0)

### Repro steps

Paket version 5.241.2

1. `dotnet new console`
1. `paket convert-from-nuget`
1. `paket add --project project.csproj Microsoft.Management.Infrastructure`

### Expected behavior

Microsoft.Management.Infrastructure resolved to 2.0

### Actual behavior

Microsoft.Management.Infrastructure resolved to 4.0:

```
NUGET
  remote: https://api.nuget.org/v3/index.json
    Microsoft.Management.Infrastructure (4.0)
``` 

### Known workarounds

Pin version in paket.dependencies

**Comment by forki** [comment]
Yes. If no other version is available in the range then we fallback to unlisted

**Comment by theimowski** [comment]
Is that by design? If so why?

**Comment by forki** [comment]
The original case was that a lot of people (including MS) released packages
as unlisted before they hit 1.0. Basically this from a time when SemVer
wasn't popular at all. And prerelease versions weren't really used.
So in order to make it work we had to fallback to unlisted versions.

This time is probably long gone and I assume you run into a new edge case.
Tbh I assume it would be fine to deprecate this fallback in v6

Tomasz Heimowski <notifications@github.com> schrieb am Mi., 18. Dez. 2019,
13:03:

**Comment by theimowski** [comment]
I'd opt to change that in v6 yeah. 

Not sure about other users, but me as a Paket user if I needed a prerelease version, I'd explicitly set it. 
So resolving to an unlisted version without any warning might be quite an issue - I'd hardly even notice if I didn't `git diff` on paket.lock

**Comment by matthid** [comment]
Strange for me paket usually skips unlisted versions in the resolution if there are no conditions which force it to use it (for example >=3 in this scenario). Maybe this is just a bug in `paket add`?

**Comment by forki** [comment]
Yes. It should only use unlisted if there is no listed version available.
